src/assets/scripts/client/canvas/CanvasController.js

```diff
@@ -1009,16 +1029,15 @@ export default class CanvasController {
                 break;
         }
 
-        const alerts = aircraftModel.hasAlerts();
-        const aircraftCanvasPosition = CanvasStageModel.translatePostionModelToPreciseCanvasPosition(aircraftModel.relativePosition);
+        const aircraftCanvasPosition = CanvasStageModel.translatePostionModelToPreciseCanvasPosition(
+            aircraftModel.relativePosition
+        );
 
         cc.translate(aircraftCanvasPosition.x, aircraftCanvasPosition.y);
 
         this._drawAircraftVectorLines(cc, aircraftModel);
-
-        if (aircraftModel.notice || alerts[0]) {
-            this._drawAircraftRings(cc, aircraftModel);
-        }
+        this._drawAircraftHalo(cc, radarTargetModel);
+        this._drawAircraftConflictRings(cc, radarTargetModel);
```

Comment:
@erikquinn I changed the order to ensure that violations are visible. i.e. violations are rendered on top of the halo (not the other way round).

Reply:
It was intentional at the time, as thats often why I would have a halo up, but after thinking about it, I agree that the violation should be drawn on top instead of hidden by the halo.